compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -665,46 +665,42 @@ impl<'a> AstValidator<'a> {
     /// - Non-const
     /// - Either foreign, or free and `unsafe extern "C"` semantically
     fn check_c_variadic_type(&self, fk: FnKind<'a>) {
-        let variadic_spans: Vec<_> = fk
-            .decl()
-            .inputs
-            .iter()
-            .filter(|arg| matches!(arg.ty.kind, TyKind::CVarArgs))
-            .map(|arg| arg.span)
-            .collect();
+        // The parser already rejects `...` when it is not the final parameter.
```

Comment:
It's not the parser, it's also AST validation (inside `check_decl_cvariadic_pos`)